File: Java/072.java
public class Problem072 {

    public static void main(String[] args) {
        int limit = 1_000_000;
        long[] phi = new long[limit + 1];

        for (int i = 0; i <= limit; i++) {
            phi[i] = i;
        }

        for (int i = 2; i <= limit; i++) {
            if (phi[i] == i) { // i is prime
                for (int j = i; j <= limit; j += i) {
                    phi[j] -= phi[j] / i;
                }
            }
        }

        long totalCount = 0;
        for (int i = 2; i <= limit; i++) {
            totalCount += phi[i];
        }

        System.out.println(totalCount);
    }
}

File: Rust/072.rs
fn main() {
    let limit = 1_000_000;
    let mut phi: Vec<u32> = (0..=limit as u32).collect();

    for i in 2..=limit {
        if phi[i] == i as u32 { // i is prime
            for j in (i..=limit).step_by(i) {
                phi[j] -= phi[j] / (i as u32);
            }
        }
    }

    let total_count: u64 = phi.iter().skip(2).map(|&x| x as u64).sum();

    println!("{}", total_count);
}
